### ISSUE TITLE
Omit HTML entities from `vue/no-bare-strings-in-template`

### ISSUE THREAD

**Issue by mrleblanc101** [vuejs/eslint-plugin-vue#2730]
**What rule do you want to change?**
`vue/no-bare-strings-in-template`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
New value for default option

**Please provide some example code that this change will affect:**

```vue
<span v-if="$slots.default" class="text-sm">
    <slot /><span class="ml-2 whitespace-nowrap"
        >&zwj;<UHelpTooltip v-if="help" :content="help" class="align-middle"
    /></span>
</span>
```

**What does the rule currently do for this code?**
> Unexpected non-translated string used.

**What will the rule do after it's changed?**
No error reported

**Additional context**

**Comment by waynzh** [comment]
There are some challenges for supporting a complete list of HTML entities. There are many HTML entities, and including all entities will make the `allowlist` very long.

IMO, the original intention of the rule design is to allow common punctuation marks and special characters, not for all. Maybe a more feasible approach could be to add only the HTML entities you actually use in your project to the `allowlist` option. If you frequently use various HTML entities, you may need to consider disabling this rule for specific code segments, or bypassing the check in other ways, like `{{ '&zwj;' }}`.

**Comment by waynzh** [comment]
I'll close this issue for now, since [v10.1.0](https://github.com/vuejs/eslint-plugin-vue/releases/tag/v10.1.0) supports regexp.
